modules/core/src/main/java/org/apache/ignite/internal/processors/cluster/GridClusterStateProcessor.java

```diff
@@ -1654,6 +1659,14 @@ public boolean isBaselineAutoAdjustEnabled() {
         return distributedBaselineConfiguration.isBaselineAutoAdjustEnabled();
     }
 
+    /**
+     * @return {@code False} if cluster deactivation can erase user's data and objects.
+     * {@code True} If deactivation is safe.
+     */
+    public boolean isDeactivationSafe() {
```

Comment:
Please, use `ctx.cache().cacheDescriptors()` here and remove `GridCacheProcessor#inMemoryCaches` method.

```
        return ctx.cache().cacheDescriptors().values().stream()
            .filter(desc -> !isPersistentCache(desc.cacheConfiguration(), ctx.config().getDataStorageConfiguration()))
            .findFirst()
            .isPresent();
```